docs/user/secure-gateways.md

```diff
@@ -0,0 +1,214 @@
+```
+
+## Testing
+
+### Clusters without External Loadbalancer Support
+
+Port forward to the Envoy service:
+
+```shell
+kubectl -n envoy-gateway-system port-forward service/envoy-default-eg 8043:8443 &
+```
+
+Curl the example app through Envoy proxy:
+
+```shell
+curl -v -HHost:www.example.com --resolve "www.example.com:8043:127.0.0.1" \
+--cacert example.com.crt https://www.example.com:8043/get
+```
+
+You can replace `get` with any of the supported [httpbin methods][httpbin_methods].
+
+### Clusters with External Loadbalancer Support
+
+Get the External IP of the Gateway:
+```shell
+export GATEWAY_HOST=$(kubectl get gateway/eg -o jsonpath='{.status.addresses[0].value}')
+```
+
+Curl the example app through the Gateway:
+
+```shell
+curl -v -HHost:www.example.com --resolve "www.example.com:8443:${GATEWAY_HOST}" \
+--cacert example.com.crt https://www.example.com:8443/get
+```
+
+You can replace `get` with any of the supported [httpbin methods][httpbin_methods].
+
+## Multiple HTTPS Listeners
+
+Due to [Issue 520][], multiple HTTP listeners must use different port numbers. For example:
+
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: eg
+spec:
+  gatewayClassName: eg
+  listeners:
+    - name: http
+      protocol: HTTP
+      port: 8080
+    - name: https
+      protocol: HTTPS
+      port: 8443
+      tls:
+        mode: Terminate
+        certificateRefs:
+          - kind: Secret
+            group: ""
+            name: example-cert
+    - name: https-2
+      protocol: HTTPS
+      port: 8444
+      tls:
+        mode: Terminate
+        certificateRefs:
+          - kind: Secret
+            group: ""
+            name: example-cert-2
+ EOF
+```
+
+Store the previously created cert/key in Secret "example-cert-2":
+
+```shell
+kubectl create secret tls example-cert-2 --key=www.example.com.key --cert=www.example.com.crt
+```
+
+Follow the steps in the [Testing section](#testing) to test connectivity to the backend app through both Gateway
+listeners.
+
+## Cross Namespace Certificate References
+
+A Gateway can be configured to reference a certificate in a different namespace. This is allowed by a [ReferenceGrant][]
+created in the target namespace. Without the ReferenceGrant, the cross-namespace reference would be invalid.
+
+To demonstrate cross namespace certificate references, create a ReferenceGrant that allows Gateways from the "default"
+namespace to reference Secrets in the "envoy-gateway-system" namespace:
+
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: ReferenceGrant
+metadata:
+  name: example
+  namespace: envoy-gateway-system
+spec:
+  from:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    namespace: default
+  to:
+  - group: ""
+    kind: Secret
+EOF
+```
+
+Delete the previously created Secret:
+
+```shell
+kubectl delete secret/example-cert
+```
```

Comment:
It appears that right after this step, the Listener is still "Ready" and still serves traffic using the now-deleted TLS cert -- possibly fixed by https://github.com/envoyproxy/gateway/pull/579?

Reply:
Thanks for finding this bug. It appears that ^ PR does not resolve this issue. I'm working through a fix that I will send a separate PR to fix.

Reply:
This is now fixed and doc updated.